markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -144,6 +153,16 @@ library PerpsMarket {
         self.size = (self.size + MathUtil.abs(newPosition.size)) - MathUtil.abs(oldPositionSize);
         self.skew += newPosition.size - oldPositionSize;
         oldPosition.updatePosition(newPosition);
+        int128 sizeDelta = newPosition.size - oldPositionSize;
+        // TODO add current market debt
+        emit MarketUpdated(
+            self.id,
+            self.skew,
+            self.size,
+            sizeDelta,
+            currentFundingRate(self),
+            currentFundingVelocity(self)
+        );
```

Comment:
maybe return this data as a tuple then emit the MarketUpdated event in async order settlement

Reply:
Yep that make sense if we move to `IMarketEvents`